google/auth/impersonated_credentials.py

```diff
@@ -433,6 +433,7 @@ def refresh(self, request):
             headers=headers,
             data=json.dumps(body).encode("utf-8"),
         )
+        response.raise_for_status()
```

Comment:
Could you write a test for this?

Reply:
@clundin25, I've attempted to write a test. Let me know if you'd like me to approach it differently.

Reply:
`refresh` method normally throws `RefreshError` (https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/exceptions.py#L35), could you maybe check the status code and raise a `RefreshError` instead?